LEGO1/omni/include/mxregion.h

```diff
@@ -1,34 +1,534 @@
+	virtual MxRect32* Prev();                 // vtable+0x30
+	virtual MxRect32* Head(MxRect32& p_rect); // vtable+0x14
+	virtual MxRect32* Tail(MxRect32& p_rect); // vtable+0x1c
+	virtual MxRect32* Next(MxRect32& p_rect); // vtable+0x24
+	virtual MxRect32* Prev(MxRect32& p_rect); // vtable+0x2c
+
+	// FUNCTION: LEGO1 0x100c4070
+	virtual MxRect32* GetRect() { return m_rect; } // vtable+0x34
+
+	// FUNCTION: LEGO1 0x100c4080
+	virtual MxBool Valid() { return m_rect != NULL; } // vtable+0x38
+
+	virtual void Reset(); // vtable+0x3c
+
+	// SYNTHETIC: LEGO1 0x100c4090
+	// MxRegionCursor::`scalar deleting destructor'
+};
+
+#ifdef REGION_SANITY_CHECK
+
+class MxRectIntersection {
+protected:
+	MxRect32 m_rect;
+	MxS32 m_numRects;
+
+public:
+	MxRect32& GetRect() { return m_rect; }
+	void SetRect(MxRect32& p_rect) { m_rect = p_rect; }
+	MxS32 GetNumRects() { return m_numRects; }
+	void SetNumRects(MxS32 p_numRects) { m_numRects = p_numRects; }
+};
+
+class MxRectIntersectionList : public MxPtrList<MxRectIntersection> {
+public:
+	MxRectIntersectionList() : MxPtrList<MxRectIntersection>(TRUE) {}
+};
+
+class MxRectIntersectionListCursor : public MxPtrListCursor<MxRectIntersection> {
+public:
+	MxRectIntersectionListCursor(MxRectIntersectionList* p_list) : MxPtrListCursor<MxRectIntersection>(p_list) {}
+};
+
+class MxRegionSanityCheck {
+protected:
+	MxRectIntersectionList* m_rectIntersectionList;
 
-private:
-	MxRegionTopBottomList* m_list; // 0x08
-	MxRect32 m_rect;               // 0x0c
+public:
+	MxRegionSanityCheck();
+	~MxRegionSanityCheck();
+	void Reset() { m_rectIntersectionList->Delete(); }
+	void AddRect(MxRect32& p_rect);
+	MxS32 CalculateArea();
 };
 
-#endif // MXREGION_H
+#endif
+
+// TEMPLATE: LEGO1 0x100c32e0
+// TEMPLATE: BETA10 0x1014ac30
+// MxCollection<MxSpan *>::Compare
+
+// TEMPLATE: LEGO1 0x100c32f0
+// TEMPLATE: BETA10 0x1014adf0
+// MxCollection<MxSpan *>::~MxCollection<MxSpan *>
+
+// TEMPLATE: LEGO1 0x100c3340
+// TEMPLATE: BETA10 0x1014ae90
+// MxCollection<MxSpan *>::Destroy
+
+// TEMPLATE: LEGO1 0x100c3350
+// TEMPLATE: BETA10 0x1014aea0
+// MxList<MxSpan *>::~MxList<MxSpan *>
+
+// TEMPLATE: LEGO1 0x100c33e0
+// TEMPLATE: BETA10 0x1014af50
+// MxPtrList<MxSpan>::Destroy
+
+// TEMPLATE: LEGO1 0x100c3480
+// TEMPLATE: BETA10 0x1014afd0
+// MxPtrList<MxSpan>::~MxPtrList<MxSpan>
+
+// SYNTHETIC: LEGO1 0x100c34d0
+// SYNTHETIC: BETA10 0x1014b030
+// MxCollection<MxSpan *>::`scalar deleting destructor'
+
+// SYNTHETIC: LEGO1 0x100c3540
+// SYNTHETIC: BETA10 0x1014b070
+// MxList<MxSpan *>::`scalar deleting destructor'
+
+// SYNTHETIC: LEGO1 0x100c35f0
+// SYNTHETIC: BETA10 0x1014b130
+// MxPtrList<MxSpan>::`scalar deleting destructor'
+
+// SYNTHETIC: LEGO1 0x100c3be0
+// SYNTHETIC: BETA10 0x1014b600
+// MxSpanListCursor::`scalar deleting destructor'
+
+// TEMPLATE: LEGO1 0x100c3c50
+// TEMPLATE: BETA10 0x1014b640
+// MxPtrListCursor<MxSpan>::~MxPtrListCursor<MxSpan>
+
+// SYNTHETIC: LEGO1 0x100c3ca0
```

Comment:
Isn't this a TEMPLATE?

Reply:
Technically it's both synthetic and template. I'm not sure if we decided what takes preference in cases like these, I think it's relatively inconsistent